### PR TITLE
Improve websocket connection events

### PR DESCRIPTION
**Pull request by josuemontano**
## Change description

The `wakeup` event is fired when the websocket connection is restored after the `suspended` and `disconnected` ably events. However, this does not reflect well the real state of the connection. Check [Ably docs](https://ably.com/docs/realtime/connection) for further information about the events fired by Ably.

- [x] Trigger the `disconnected` event when the connection is lost at least for 2 mins.
- [x] Trigger the `reconnected` event when the computer awakes from sleep.
- [x] Trigger the `reconnected` event when the connection is recovered, even if it is for less than 2 mins.

## Type of change
- [x] Bug fix (fixes an issue)
- [ ] New feature (adds functionality)

### Development

- [x] Lint rules pass locally
- [x] Application changes have been tested thoroughly
- [ ] Automated tests covering modified code pass

### Security

- [x] Security impact of change has been considered
- [x] Code follows company security practices and guidelines

### Code review 

- [x] Pull request has a descriptive title and context useful to a reviewer. Screenshots or screencasts are attached as necessary
- [x] "Ready for review" label attached and reviewers assigned
- [ ] Changes have been reviewed by at least one other contributor
- [ ] Pull request linked to task tracker where applicable

### CODE REVIEW

File: src/lib/ajax.ts [hunk 1]
```diff
@@ -28,6 +28,7 @@ export function buildAPIHeaders() {
   const headers = {
     'X-MAGICBELL-CLIENT-ID': clientId,
     'X-MAGICBELL-API-KEY': apiKey,
+    'X-MagicBell-Client': 'ReactHeadless/4.0.0', // @todo: Get value from package.json
```
Comment (smeijer) [inline src/lib/ajax.ts:31]: Should we fix this in this PR? The version provides false information if we forget to bump it on a package update. Just a thought; we can add constant like `__CLIENT_VERSION__` that we'll replace in our build step, using [`microbundle --define`](https://github.com/developit/microbundle#defining-build-time-constants).

Also, would it make sense to keep the client name the same as the package name, and the key uppercase like the other headers?

```
X-MAGICBELL-CLIENT: `@magicbell/react-headless@{VERSION}`
```

**Review by smeijer — APPROVED** [review]

File: src/lib/ajax.ts (hunk 1 above)
Comment (josuemontano) [reply src/lib/ajax.ts:31]: Probably, let's list all libraries and create a standard. Not every SDK has the `@magicbell` prefix, that's why I didn't use it here.

> Also, would it make sense to keep the client name the same as the package name, and the key uppercase like the other headers?
